Modules/_zstd/compressor.c

```diff
@@ -690,9 +690,84 @@ _zstd_ZstdCompressor_flush_impl(ZstdCompressor *self, int mode)
     return ret;
 }
 
+
+/*[clinic input]
+_zstd.ZstdCompressor.set_pledged_input_size
+
+    size: object
+        The size of the uncompressed data to be provided to the compressor.
+    /
+
+Set the uncompressed content size to be written into the frame header.
+
+This method can be used to ensure the header of the frame about to be written
+includes the size of the data, unless CompressionParameter.content_size_flag is
+set to False. If .last_mode != .FLUSH_FRAME, then a RuntimeError is raised.
+
+It is important to ensure that the pledged data size matches the actual data
+size. If they do not match the compressed output data may be corrupted and the
+final chunk written may be lost.
+[clinic start generated code]*/
+
+static PyObject *
+_zstd_ZstdCompressor_set_pledged_input_size_impl(ZstdCompressor *self,
+                                                 PyObject *size)
+/*[clinic end generated code: output=2fb57610ee586d42 input=272a7881a3526dd1]*/
+{
+    uint64_t pledged_size;
+    size_t zstd_ret;
+    PyObject *ret;
+
+    /* Get size value */
+    if (size == Py_None) {
+        pledged_size = ZSTD_CONTENTSIZE_UNKNOWN;
+    } else {
+        pledged_size = PyLong_AsUnsignedLongLong(size);
+        if (pledged_size == (uint64_t)-1 && PyErr_Occurred()) {
+            if (PyErr_ExceptionMatches(PyExc_OverflowError)) {
+                PyErr_Format(PyExc_ValueError,
+                             "size argument should be a positive int less "
+                             "than %ull", ZSTD_CONTENTSIZE_ERROR);
```

Comment:
Should not this exception also be raised when pledged_size >= ZSTD_CONTENTSIZE_ERROR?

Reply:
This is now handled in the argument clinic converter, but effectively, it already is handled by the fact that `ZSTD_CONTENTSIZE_ERROR == 0ULL -1`

Reply:
Added these and a couple of other tests (e.g. 0)